src/Processors/Formats/Impl/Parquet/ParquetLeafColReader.cpp

```diff
@@ -488,17 +455,114 @@ void ParquetLeafColReader<TColumn>::readPageV1(const parquet::DataPageV1 & page)
         case parquet::Encoding::DELTA_BINARY_PACKED:
         case parquet::Encoding::DELTA_LENGTH_BYTE_ARRAY:
         case parquet::Encoding::DELTA_BYTE_ARRAY:
-            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unsupported encoding: {}", page.encoding());
+            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unsupported encoding: {}", enconding_type);
 
         default:
-          throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unknown encoding type: {}", page.encoding());
+            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unknown encoding type: {}", enconding_type);
     }
 }
 
 template <typename TColumn>
-void ParquetLeafColReader<TColumn>::readPageV2(const parquet::DataPageV2 & /*page*/)
+void ParquetLeafColReader<TColumn>::readPageV1(const parquet::DataPageV1 & page)
 {
-    throw Exception(ErrorCodes::NOT_IMPLEMENTED, "read page V2 is not implemented yet");
+    static parquet::LevelDecoder repetition_level_decoder;
+
+    cur_page_values = page.num_values();
+
+    // refer to: VectorizedColumnReader::readPageV1 in Spark and LevelDecoder::SetData in column_reader.cc
+    if (page.definition_level_encoding() != parquet::Encoding::RLE && col_descriptor.max_definition_level() != 0)
+    {
+        throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unsupported encoding: {}", page.definition_level_encoding());
+    }
+    const auto * buffer =  page.data();
+    auto max_size = page.size();
+
+    if (col_descriptor.max_repetition_level() > 0)
+    {
+        auto rep_levels_bytes = repetition_level_decoder.SetData(
+            page.repetition_level_encoding(), col_descriptor.max_repetition_level(), 0, buffer, max_size);
+        buffer += rep_levels_bytes;
+        max_size -= rep_levels_bytes;
+    }
+
+    assert(col_descriptor.max_definition_level() >= 0);
+    std::unique_ptr<RleValuesReader> def_level_reader;
+    if (col_descriptor.max_definition_level() > 0)
+    {
+        auto bit_width = arrow::bit_util::Log2(col_descriptor.max_definition_level() + 1);
+        auto num_bytes = ::arrow::util::SafeLoadAs<int32_t>(buffer);
+        auto bit_reader = std::make_unique<arrow::bit_util::BitReader>(buffer + 4, num_bytes);
```

Comment:
Again not related to this PR: this is missing range checks: (1) that buffer has at least 4 bytes for the `SafeLoadAs<int32_t>`, (2) that `4 + num_bytes <= max_size`. Can you please add them along the way? Same checks would be needed for the `SetData` replacement as in comment above.